tests/build/run_cython_not_in_cwd.srctree

```diff
@@ -22,35 +23,24 @@ import subprocess
 python = sys.executable
 
 def cython(cmdstr, cwd=None):
-    argv = [sys.executable, '-m', 'cython'] + cmdstr.split()
+    argv = [python, '-m', 'cython'] + cmdstr.split()
     subprocess.run(argv, check=True, cwd=cwd)
 
-def grep(fname, pattern):
+def grep_paths(fname, pattern):
     with open(fname) as fin:
-        return [line for line in fin if pattern in line]
-
-cython('-3 src/foo/bar.pyx -o build/src/foo/bar1.pyx.c')
-cython('-3 ../src/foo/bar.pyx -o src/foo/bar2.pyx.c', cwd='build')
-
-res1 = grep('build/src/foo/bar1.pyx.c', 'bar.pyx')
-res2 = grep('build/src/foo/bar2.pyx.c', 'bar.pyx')
-
-assert res1 == [
-    # Running cython on a file in CWD uses some paths relative to CWD.
-    '  "src/foo/bar.pyx",\n',
-    'static const char __pyx_k_src_foo_bar_pyx[] = "src/foo/bar.pyx";\n',
-    '/* "foo/bar.pyx":2\n',
-    '  /* "foo/bar.pyx":3\n',
-    '  /* "foo/bar.pyx":2\n',
-    '  /* "foo/bar.pyx":2\n'
-], '\n' + ''.join(res1)
-
-assert res2 == [
-    # When not in CWD all paths are relative to sys.path.
-    '  "foo/bar.pyx",\n',
-    'static const char __pyx_k_foo_bar_pyx[] = "foo/bar.pyx";\n',
-    '/* "foo/bar.pyx":2\n',
-    '  /* "foo/bar.pyx":3\n',
-    '  /* "foo/bar.pyx":2\n',
-    '  /* "foo/bar.pyx":2\n',
-], '\n' + ''.join(res2)
+        for line in fin:
+            if pattern in line:
+                [part] = [p for p in line.split() if pattern in p]
+                yield part.rstrip(',\n;:01234566789').strip('"')
+
+cython('-3 src/pkg/mod.pyx -o build/src/pkg/mod1.pyx.c')
+cython('-3 ../src/pkg/mod.pyx -o src/pkg/mod2.pyx.c', cwd='build')
+
+res1 = grep_paths('build/src/pkg/mod1.pyx.c', 'mod.pyx')
+res2 = grep_paths('build/src/pkg/mod2.pyx.c', 'mod.pyx')
+
+# Running cython on a file in CWD uses some paths relative to CWD.
+assert set(res1) == {"src/pkg/mod.pyx", "pkg/mod.pyx"}, str(res1)
```

Comment:
The way you wrote this, `str(res1)` will print something like `<generator  …>` on failure. You should collect the generator result into a `set()` right when calling it.